lib/src/data/tutor_availability.dart

```diff
@@ -0,0 +1,55 @@
+import "utils.dart";
+
+extension type TutorID(String value) {}
+extension type CourseID(String value) {}
+
+class TutorAvailability {
+  final TutorID tutorId;
+  final String department;
+  final List<CourseID> availableCourses;
+  final String tutorName;
+  final String appointmentType;
+  final DateTime appointmentDateTime;
+  final int? maxStudents;
+  final String location;
+  final String tutorContact;
+  final String? note;
+
+  TutorAvailability({
+    required this.tutorId,
+    required this.department,
+    required this.availableCourses,
+    required this.tutorName,
+    required this.appointmentType,
+    required this.appointmentDateTime,
+    required this.maxStudents,
+    required this.location,
+    required this.tutorContact,
+    this.note,
+  });
+
+  TutorAvailability.fromJson(Json json)
+      : tutorId = TutorID(json["tutorId"]),
+        department = json["department"],
+        availableCourses = json["availableCourses"]?.cast<String>() ?? [],
+        tutorName = json["tutorName"],
+        appointmentType = json["appointmentType"],
+        appointmentDateTime = DateTime.parse(json["appointmentDateTime"]),
+        maxStudents = json["maxStudents"],
+        location = json["location"],
+        tutorContact = json["tutorContact"],
+        note = json["note"];
+
+  Json toJson() => {
+        "tutorId": tutorId.value,
+        "department": department,
+        "availableCourses": availableCourses,
+        "tutorName": tutorName,
+        "appointmentType": appointmentType,
+        "appointmentDateTime": appointmentDateTime,
+        "maxStudents": maxStudents,
+        "location": location,
+        "tutorContact": tutorContact,
+        "note": note,
+      };
```

Comment:
This is probably from `dart format`, but un-indent these back 1 level, and you can put the `:` from the constructor on the same line, like 
```dart
  TutorAvailability.fromJson(Json json) : 
    tutorId = // ...
```